File: interviewbit-bit-manipulation-reverse-bits.py
class Solution:
    # @param A : unsigned integer
    # @return an unsigned integer
    def reverse(self, A):
        digits = [x for x in map(int, list(bin(A)[2:]))][::-1]
        digits += [0] * (32 - len(digits))
        return int("".join(map(str, digits)), 2)

File: interviewbit-bit-manipulation-reverse-bits.java
public class Solution {
    public long reverse(long a) {
        long rev = 0;
        for (int i = 0; i < 32; i++) {
            rev <<= 1;
            rev += a & 1;
            a >>= 1;
        }
        return rev;
    }
}
